core/src/main/resources/hudson/model/AbstractItem/rename2.jelly

```diff
@@ -22,43 +22,21 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 -->
 
-<!-- Rename project -->
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:i="jelly:fmt">
-  <l:layout>
+  <l:layout title="${%Rename}">
     <st:include page="sidepanel.jelly" />
     <l:main-panel>
-      <j:set var="newName" value="${request.getParameter('newName')}" />
-      <j:set var="checkIfNameIsUsed" value="${it.checkIfNameIsUsed(newName)}" />
-      
-      <j:choose> 
-        <j:when test="${it.isBuilding()}">
-          ${%noRenameWhileBuilding}
-          <j:if test="${request.referer.endsWith('/configure')}">
-            <br/> ${%configWasSaved}
-          </j:if>
-        </j:when>
-        
-        <j:when test="${checkIfNameIsUsed != null and !it.name.equalsIgnoreCase(newName)}" >    
-          <j:if test="${checkIfNameIsUsed}" >
-            ${%newNameInUse(newName)}
-          </j:if>  
-          <j:if test="${!checkIfNameIsUsed}" >
-            ${%newNameNotValid(newName)}
-          </j:if>
-          <j:if test="${request.referer.endsWith('/configure')}">
-            <br/> ${%configWasSaved}
-          </j:if>
-        </j:when>
-        
-        <j:otherwise>
-          <form method="post" action="doRename">
-            ${%description(it.name, newName)}
-            <input type="hidden" name="newName" value="${newName}" />
-            <f:submit value="${%Yes}" />
-          </form>
-        </j:otherwise>
-      </j:choose>
+      <h1><l:icon class="icon-notepad icon-xlg"/> ${%DescribeRename(it.pronoun, it.name)}</h1>
+      <f:form method="post" action="doRename2" name="config" tableClass="config-table scrollspy">
+        <f:entry title="${%NewName}">
+          <f:textbox name="newName" value="${it.name}" autocomplete="on" checkUrl="checkNewName" checkDependsOn="newName"/>
```

Comment:
I think using `checkUrl` and `checkDependsOn` like this without a descriptor is a hack, but I'm not sure. Is there a preferred method of doing this?

Reply:
I agree it looks a little hacky but this definitely falls into "how the heck do you Jenkins", so hopefully @oleg-nenashev can answer.

Reply:
Probably you can just use

```xml
<f:entry field="name" title="${%NewName}">
    <f:textbox/>
</f:entry>
```

and name web methods appropriately (`doCheckName` and have the submission URL take a `name` parameter).

Reply:
I played around with those changes but couldn't get it to work. The submission URL doesn't get the value unless I add `name="blah"`, and the form validation doesn't trigger.